azkaban-common/src/main/java/azkaban/project/ProjectManager.java

```diff
@@ -86,12 +98,115 @@ public ProjectManager(ProjectLoader loader, Props props) {
     new XmlValidatorManager(prop);
     loadAllProjects();
     loadProjectWhiteList();
+
+    CleanerThread cleanerThread = new CleanerThread();
+    cleanerThread.start();
+  }
+
+  /**
+   * Cleaner thread to clean projects. Runs every day.
+  **/
+  private class CleanerThread extends Thread {
+
+    // runs every day
+    private static final long CLEANER_THREAD_WAIT_INTERVAL_MS = 24 * 60 * 60 * 1000;
+    private static final String DELETER = "azkaban";
+
+    private boolean shutdown = false;
+    private long lastProjectFileCleanTime = -1;
+    private long lastProjectCleanTime = -1;
+
+    public CleanerThread() {
+      this.setName("ProjectManager-Cleaner-Thread");
+      logger.info("Starting cleaner thread: " + this.getName());
+    }
+
+    @SuppressWarnings("unused")
+    public void shutdown() {
```

Comment:
Who is calling this ? Worst case we need to register a shutdownhook isn't it ?

Reply:
Currently, it is not getting called from anywhere, there was already a cleanerThread impl in FlowRunnerManager, I followed the same. This shutdown function was added in some other functions as well there also it is not getting used, in some cases it was used in only testcases